src/views/docs-view/__tests__/multiple-remark-plugins.test.ts

```diff
@@ -4,42 +4,62 @@ import { rewriteTutorialLinksPlugin } from 'lib/remark-plugins/rewrite-tutorial-
 import remarkPluginAdjustLinkUrls from 'lib/remark-plugin-adjust-link-urls'
 import { getProductUrlAdjuster } from '../utils/product-url-adjusters'
 
-describe('multiple remark plugins', () => {
-	let waypointProduct
-	let productUrlAdjuster
+const URL_SUFFIXES = ['', '?param=value', '#heading', '?param=value#heading']
 
-	beforeAll(() => {
-		waypointProduct = cachedGetProductData('waypoint')
-		productUrlAdjuster = getProductUrlAdjuster(waypointProduct)
+const testEachCase = (productUrlAdjuster, testCases) => {
+	const allTestCases = []
+	testCases.forEach(({ input, expected }) => {
+		URL_SUFFIXES.forEach((suffix) => {
+			allTestCases.push({
+				input: `${input}${suffix}`,
+				expected: `${expected}${suffix}`,
+			})
+		})
 	})
 
-	describe('rewriteTutorialLinksPlugin, then remarkPluginAdjustLinkUrls', () => {
-		test.each([
-			['[Waypoint](https://waypointproject.io/)', '[Waypoint](/waypoint)'],
-			['[Waypoint](/waypoint)', '[Waypoint](/waypoint)'],
-			[
-				'[Waypoint Tutorials](https://learn.hashicorp.com/waypoint)',
-				'[Waypoint Tutorials](/waypoint/tutorials)',
-			],
-			[
-				'[Waypoint Tutorials](/waypoint/tutorials)',
-				'[Waypoint Tutorials](/waypoint/tutorials)',
-			],
-			[
-				'[Some other link](https://kubernetes.io/docs/home/)',
-				'[Some other link](https://kubernetes.io/docs/home/)',
-			],
-		])('%s -> %s', async (linkToProcess, expectedValue) => {
+	test.each(allTestCases)(
+		'$input -> $expected',
+		async ({ input, expected }) => {
 			const result = await remark()
 				.use(rewriteTutorialLinksPlugin, {
 					contentType: 'docs',
 					tutorialMap: {},
 				})
-				.use(remarkPluginAdjustLinkUrls, { urlAdjustFn: productUrlAdjuster })
-				.process(linkToProcess)
+				.use(remarkPluginAdjustLinkUrls, {
+					urlAdjustFn: productUrlAdjuster,
+				})
+				.process(`[test](${input})`)
+			expect(result.contents).toMatch(`[test](${expected})`)
+		}
+	)
+}
 
-			// The link should be unchanged
-			expect(result.contents).toMatch(expectedValue)
-		})
+describe('multiple remark plugins', () => {
```

Comment:
Nice, man the multiple remark plugins is nasty to track down this stuff! well done